app/assets/javascripts/admin/addon/components/admin-config-area-sidebar-experiment.js

```diff
@@ -84,12 +84,16 @@ export default class AdminConfigAreaSidebarExperiment extends Component {
           return;
         }
 
+        // Using the private `_routerMicrolib` is not ideal, but Ember doesn't provide
+        // any other way for us to easily check for route validity.
         try {
           this.router._router._routerMicrolib.recognizer.handlersFor(
```

Comment:
I would have also liked to validate the custom links added by plugins to the admin sidebar in this way, but sadly this cannot be done inside initializers in ember -- the router is not set up yet completely.

Also when using the old method of plugin outlets, the developer can put in whatever route they want and there's no additional validation, the page will just error if you get the route incorrect, so I think I am fine to accept this:

```
<NavItem @route="upgradeBLAHBLAH" @label="admin.docker.upgrade_tab" />
```